iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -435,6 +486,70 @@ public override async Task OpenAsync(CancellationToken cancellationToken)
             }
         }
 
+        private void StartLoopAsync(DateTime refreshOn, CancellationToken cancellationToken)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, nameof(StartLoopAsync));
+
+            if (_loopCancellationTokenSource == null
+                || _refreshLoop == null)
+            {
+                (this as IAmqpAuthenticationRefresher)?.StopLoop();
+            }
+
+            _loopCancellationTokenSource = new CancellationTokenSource();
+            _refreshLoop = RefreshLoopAsync(refreshOn, cancellationToken);
+
+            if (Logging.IsEnabled)
+                Logging.Exit(this, nameof(StartLoopAsync));
+        }
+
+        private async Task RefreshLoopAsync(DateTime refreshesOn, CancellationToken cancellationToken)
+        {
+            TimeSpan waitTime = refreshesOn - DateTime.UtcNow;
+
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Info(this, refreshesOn, $"Before {nameof(RefreshLoopAsync)} with wait time {waitTime}.");
+
+                if (waitTime > TimeSpan.Zero)
+                {
+                    await Task.Delay(waitTime, cancellationToken).ConfigureAwait(false);
+                }
+
+                if (!cancellationToken.IsCancellationRequested)
```

Comment:
Is this `if` check necessary? If it had canceled, `Task.Delay` would have thrown. If after, shouldn't `RefreshTokenAsync` throw?